pkg/local_workflows/filter_workflow.go

```diff
@@ -0,0 +1,122 @@
+	flags := pflag.NewFlagSet(FilterFindingsWorkflowName, pflag.ExitOnError)
+	_, err := engine.Register(WORKFLOWID_FILTER_FINDINGS, workflow.ConfigurationOptionsFromFlagset(flags), filterFindingsEntryPoint)
+
+	return err
+}
+
+type FindingsFilterFunc func(local_models.FindingResource) bool
+
+func contains(slice []string, item string) bool {
+	for _, s := range slice {
+		if s == item {
+			return true
+		}
+	}
+	return false
+}
+
+func filterSeverityASC(original []string, severityMinLevel string) []string {
+	if severityMinLevel == "" {
+		return original
+	}
+
+	minLevelPointer := slices.Index(original, severityMinLevel)
+
+	if minLevelPointer >= 0 {
+		return original[minLevelPointer:]
+	}
+
+	return original
+}
+
+func filterFindingsBySeverity(severityThreshold string, severityOrder []string) FindingsFilterFunc {
+	return func(finding local_models.FindingResource) bool {
+		allowed_severities := filterSeverityASC(severityOrder, severityThreshold)
+
+		return contains(allowed_severities, string(finding.Attributes.Rating.Severity.Value))
+	}
+}
+
+func filterFindings(findingsModel local_models.LocalFinding, filter []FindingsFilterFunc) local_models.LocalFinding {
+	filteredFindings := local_models.LocalFinding{}
+	for _, finding := range findingsModel.Findings {
+		match := true
+		for _, filter := range filter {
+			if match {
+				match = filter(finding)
+			}
+		}
+		if match {
+			filteredFindings.Findings = append(filteredFindings.Findings, finding)
+		}
+	}
+	return filteredFindings
+}
+
+func filterFindingsEntryPoint(invocationCtx workflow.InvocationContext, input []workflow.Data) (output []workflow.Data, err error) {
+	config := invocationCtx.GetConfiguration()
+	logger := invocationCtx.GetEnhancedLogger()
+	ff_transform_enabled := config.GetBool(configuration.FF_TRANSFORMATION_WORKFLOW)
+	output = []workflow.Data{}
+
+	logger.Println("Workflow Name:", FilterFindingsWorkflowName)
+
+	if !ff_transform_enabled {
+		return input, nil
+	}
+
+	var findingsModel local_models.LocalFinding
+
+	for _, data := range input {
+		if strings.HasPrefix(data.GetContentType(), content_type.LOCAL_FINDING_MODEL) {
+			findingsBytes, ok := data.GetPayload().([]byte)
+			if !ok {
+				logger.Err(nil).Msg("invalid payload type")
+				return input, fmt.Errorf("invalid payload type")
+			}
+			unmarshalErr := json.Unmarshal(findingsBytes, &findingsModel)
+			if unmarshalErr != nil {
+				logger.Err(unmarshalErr).Msg(unmarshalErr.Error())
+				return input, unmarshalErr
+			}
+		} else {
+			output = append(output, data)
+		}
+	}
+	severityOrder := findingsModel.Summary.SeverityOrderAsc
+	filter := filterFindingsBySeverity(config.GetString(configuration.FLAG_SEVERITY_THRESHOLD), severityOrder)
+	filteredFindings := filterFindings(findingsModel, []FindingsFilterFunc{filter})
+	filteredFindingsBytes, err := json.Marshal(filteredFindings)
+	if err != nil {
+		logger.Err(err).Msg(err.Error())
+		return output, err
+	}
+	output = append(output, workflow.NewData(
```

Comment:
question: if the intention is to return a filtered list of findings, do we want to append to the existing `output`?